pmd-plsql/etc/grammar/PldocAST.jjt

```diff
@@ -2494,57 +2494,68 @@ ASTEqualityExpression EqualityExpression() #EqualityExpression(>1) :
  }
 }
 
+/**
+ * Relational operators: https://docs.oracle.com/en/database/oracle/oracle-database/18/lnpls/plsql-language-fundamentals.html#GUID-2EB964C8-9B36-4009-A45F-B340F44EE2A2
+ */
 ASTRelationalExpression RelationalExpression() #RelationalExpression(>1) :
 { PLSQLNode simpleNode = null; StringBuilder sb = new StringBuilder() ; }
 {
  (
   //AdditiveExpression() ( ( "<" | ">" | "<=" | ">=" | [<NOT>] (<IN> | <BETWEEN> | <LIKE>) ) AdditiveExpression() )*
-  (simpleNode = AdditiveExpression() ) { sb.append(simpleNode.getImage()); }
-  ( 
-   ( 
+  ( simpleNode = AdditiveExpression() ) { sb.append(simpleNode.getImage()); }
+  (
+    (
+
       (
-	(
-	    (("<" ) { sb.append("<"); } 
-	     [">"  { sb.append(">"); } ] // <> 
-	    )
-	  | (">" ) { sb.append(">"); }
-	)
-	[ ("=" ) { sb.append("="); } ]
-    
+          ("<>" { sb.append(" <> "); } )
```

Comment:
This think can probably be factored out in another production for readability, e.g
```java
void ComparisonOperator(StringBuilder sb) #void: 
{}
{
  ( "<>"
  | "<="
  | ...
  ) {sb.append(" " + getToken(0).getImage() + " ");}
}

Reply:
I wouldn't bother here at the moment - the grammar is probably mostly wrong at many places. E.g., `"="` itself would also be a relational operator, but adding this to the list changes the parser to not choose EqualityExpression later on, which then suddenly changes other (maybe unrelated) parts of the AST. I didn't look up yet whether using EqualityExpression for `"="` is right or wrong - but it's seems the way, we do at the moment, so leaving this untouched.

Reply:
Ok if I understand correctly readability is the least of our problems with the current grammar :laughing: I'll merge asap